src/pybind/mgr/cephadm/schedule.py

```diff
@@ -67,7 +68,10 @@ def __init__(self,
         self.daemons = get_daemons_func(self.service_name)
 
     def validate(self):
-        self.spec.validate()
+        try:
+            self.spec.validate()
+        except Exception as e:
+            raise ServiceSpecValidationError(str(e))
```

Comment:
any thoughts how to better pickle these exceptions so there in not a need to catch/rethrow?